kiwi/storage/subformat/vagrant_libvirt.py

```diff
@@ -83,71 +46,26 @@ def create_image_format(self):
                 box_img
             ]
         )
+        return [box_img]
 
-        metadata_json = os.sep.join([self.temp_image_dir, 'metadata.json'])
-        with open(metadata_json, 'w') as meta:
-            meta.write(self._create_box_metadata())
-
-        vagrantfile = os.sep.join([self.temp_image_dir, 'Vagrantfile'])
-        with open(vagrantfile, 'w') as vagrant:
-            vagrant.write(self._create_box_vagrantconfig())
-
-        Command.run(
-            [
-                'tar', '-C', self.temp_image_dir,
-                '-czf', self.get_target_file_path_for_format(self.image_format),
-                os.path.basename(box_img),
-                os.path.basename(metadata_json),
-                os.path.basename(vagrantfile)
-            ]
-        )
-
-    def store_to_result(self, result):
+    def get_additional_metadata(self):
+        # type: () -> Dict[str, str]
         """
-        Store result file of the vagrant format conversion into the
-        provided result instance. In this case compression is unwanted
-        because the box is already created as a compressed tarball
-
-        :param object result: Instance of Result
+        Returns a dictionary containing the virtual image format and the size
+        of the image.
         """
-        result.add(
-            key='disk_format_image',
-            filename=self.get_target_file_path_for_format(
-                self.image_format
-            ),
-            use_for_bundle=True,
-            compress=False,
-            shasum=True
-        )
-
-    def _create_box_metadata(self):
-        metadata = {
-            'provider': 'libvirt',
+        return {
             'format': 'qcow2',
             'virtual_size': format(self.vagrantconfig.get_virtualsize() or 42)
         }
-        return json.dumps(
-            metadata, sort_keys=True, indent=2, separators=(',', ': ')
-        )
-
-    def _create_box_vagrantconfig(self):
-        vagrantconfig = dedent('''
-            Vagrant::Config.run do |config|
-              config.vm.base_mac = "{mac_address}"
-            end
-            include_vagrantfile = File.expand_path(
-              "../include/_Vagrantfile", __FILE__
-            )
-            load include_vagrantfile if File.exist?(include_vagrantfile)
-        ''').strip()
-        return vagrantconfig.format(
-            mac_address=self._random_mac()
-        )
 
-    def _random_mac(self):
-        return '%02x%02x%02x%02x%02x%02x'.upper() % (
-            0x00, 0x16, 0x3e,
-            random.randrange(0, 0x7e),
-            random.randrange(0, 0xff),
-            random.randrange(0, 0xff)
-        )
+    def get_additional_vagrant_config_settings(self):
+        # type: () -> str
+        """
+        Returns settings for the libvirt provider telling vagrant to use kvm.
+        """
+        return dedent('''
+        config.vm.provider :libvirt do |libvirt|
+          libvirt.driver = "kvm"
+        end
+            ''').strip()
```

Comment:
With templating in place and the fact that the string is stripped anyway this can be imho nicer indented:

```python
return dedent('''
    config.vm.provider :libvirt do |libvirt|
      libvirt.driver = "kvm"
￼    end
''').strip()
```